### DISCUSSION TITLE
Embed "Neutral point light texture" into the engine

### DISCUSSION THREAD

**Discussion by shak2** [godotengine/godot-proposals#10146] [2D]
Neutral Point Light Texture:

[https://docs.godotengine.org/en/stable/tutorials/2d/2d_lights_and_shadows.html](https://docs.godotengine.org/en/stable/tutorials/2d/2d_lights_and_shadows.html)

As much of Godot should work out of the box as possible. Make this texture a default in engine. Users shouldn't have to go to a random page on the docs to download a random texture to get a working 2D light. Drag and drop.

Godot should "just work" as much as possible.

Actually, as a general rule, Godot should always revert to a known working default vs. throwing a warning that something is missing needs deep discussion and resolution.

**Comment by AThousandShips** [comment]
A single texture isn't necessarily enough to cover this, it depends on the situation and it has to be adjusted to the size you need, or it'll get pixelated, or waste data

As suggested in the documentation, the general purpose (or fallback) solution is to use a gradient:
> If you need different falloff, you can procedurally create a texture by assigning a New GradientTexture2D on the light's Texture property. After creating the resource, expand its Fill section and set the fill mode to Radial. You will then have to adjust the gradient itself to start from opaque white to transparent white, and move its starting location to be in the center.

**Reply by shak2** [reply]
The docs recommend that texture. So i'm thinking why not just make it the default, since you NEED something to be added to make 2d light work.

We need to make as much of Godot just work as possible, primarily through excellent defaults.

As the code base expands it will get heavier with "warnings" and incantations like the gradiantTexture2D 5 step process instead of just working. And every one of those is additional undesirable cognitive load.

**Reply by Calinou** [reply]
This was discussed elsewhere recently (I can't find it right now), but changing the behavior when no texture is applied to a PointLight2D is a behavior change that could break visuals in some existing projects. It's not a dealbreaker since this is a rare scenario to have in production, but it's something that needs to be considered.

**Reply by AThousandShips** [reply]
I'd say instead of embedding a default texture that's not guaranteed to be useful in a certain case it should be a bit easier to achieve it easily, though making a gradient texture isn't terribly difficult and you can reuse it once created
